pss/keystore.go

```diff
@@ -180,22 +181,22 @@ func (ks *KeyStore) processSym(envelope *envelope) (*receivedMessage, string, Ps
 // encapsulating the decrypted message, and the byte representation of
 // the public key used to decrypt the message.
 // It fails if decryption of message fails, or if the message is corrupted.
-func (p *Pss) processAsym(envelope *envelope) (*receivedMessage, string, PssAddress, error) {
+func (p *Pss) processAsym(pssMsg *PssMsg) (crypto.ReceivedMessage, string, PssAddress, error) {
 	metrics.GetOrRegisterCounter("pss.process.asym", nil).Inc(1)
 
-	recvmsg, err := envelope.openAsymmetric(p.privateKey)
+	recvmsg, err := p.Crypto.UnWrapAsymmetric(pssMsg.Payload, p.privateKey)
 	if err != nil {
 		return nil, "", nil, fmt.Errorf("could not decrypt message: %s", err)
 	}
 	// check signature (if signed), strip padding
-	if !recvmsg.validateAndParse() {
+	if !recvmsg.ValidateAndParse() {
 		return nil, "", nil, errors.New("invalid message")
 	}
-	pubkeyid := common.ToHex(p.Crypto.FromECDSAPub(recvmsg.Src))
+	pubkeyid := common.ToHex(p.Crypto.FromECDSAPub(recvmsg.GetSrc()))
```

Comment:
try to migrate all `common.ToHex` to `hex.EncodeToString`

Reply:
This particular case is dangerous. As `common.ToHex` prepend `0x` to the string but `hex.EncodeToString` not, there are some incompatibilites that make the tests fail. I will investigate further.